core/src/main/java/com/sequenceiq/distrox/v1/distrox/StackOperations.java

```diff
@@ -234,7 +235,9 @@ public UpgradeV4Response checkForClusterUpgrade(@NotNull NameOrCrn nameOrCrn, Lo
             boolean osUpgrade = upgradeService.isOsUpgrade(request);
             UpgradeV4Response upgradeResponse = clusterUpgradeAvailabilityService.checkForUpgradesByName(workspaceId, stackName,
                     osUpgrade);
-            clusterUpgradeAvailabilityService.filterUpgradeOptions(upgradeResponse, request);
+            if (CollectionUtils.isNotEmpty(upgradeResponse.getUpgradeCandidates())) {
+                clusterUpgradeAvailabilityService.filterUpgradeOptions(upgradeResponse, request);
+            }
```

Comment:
I am not sure this is related to the rest of the changes. If not, it might make sense to extract it to a separate PR.